migrations/000028_app_versions_table.up.sql

```diff
@@ -0,0 +1,193 @@
+BEGIN;
+
+SET search_path = public, pg_catalog;
+
+--
+-- app_versions table
+--
+CREATE TABLE IF NOT EXISTS app_versions (
+    id uuid NOT NULL DEFAULT uuid_generate_v1(),
+    app_id uuid NOT NULL,
+    version varchar(255) NOT NULL,
+    version_order int NOT NULL DEFAULT 0,
+    deleted boolean NOT NULL DEFAULT false,
+    disabled boolean NOT NULL DEFAULT false,
+    integration_data_id uuid NOT NULL,
+    integration_date timestamp without time zone,
+    edited_date timestamp without time zone DEFAULT CURRENT_TIMESTAMP,
+
+    PRIMARY KEY (id),
+    FOREIGN KEY (app_id) REFERENCES apps(id) ON DELETE CASCADE,
+    UNIQUE (app_id, version_order)
+);
+
+CREATE INDEX IF NOT EXISTS app_versions_app_id ON app_versions(app_id);
+CREATE INDEX IF NOT EXISTS app_versions_version_order ON app_versions(version_order);
+
+--
+-- Initial app_versions data
+--
+INSERT INTO app_versions (
+    app_id,
+    version,
+    deleted,
+    disabled,
+    integration_data_id,
+    integration_date,
+    edited_date
+)
+    SELECT
+        id AS app_id,
+        'Unversioned' AS version,
+        deleted,
+        disabled,
+        integration_data_id,
+        integration_date,
+        edited_date
+    FROM apps
+    ON CONFLICT DO NOTHING;
+
+--
+-- Replace the `app_id` column in the `app_steps` table
+-- with a new `app_version_id` column.
+--
+ALTER TABLE ONLY app_steps
+    ADD COLUMN IF NOT EXISTS app_version_id uuid;
+UPDATE app_steps
+    SET app_version_id = (
+        SELECT id
+        FROM app_versions
+        WHERE app_steps.app_id = app_versions.app_id
+    );
+ALTER TABLE ONLY app_steps
+    ALTER COLUMN app_version_id SET NOT NULL;
+ALTER TABLE ONLY app_steps
+    DROP COLUMN IF EXISTS app_id;
```

Comment:
Initially I was just going to rename the `app_id` column to `app_version_id`, but I would also have to drop the foreign key to `apps(id)` which is not named in the new migration scripts. So it appears I would have to do some queries into the info schemas to drop all FKs, and then re-add the required FKs for this table, instead of just dropping the named FK to `apps(id)`.

It seemed simpler to do these add/copy/drop queries instead.